src/tanks/Tancik.java

```diff
@@ -9,13 +9,21 @@
     public class Tancik extends AdvancedRobot{
 
         public Vector<Gene> genes;
+        public Vector<Gene> genesRun;
+        public Vector<Gene> genesOnScanned;
+        public Vector<Gene> genesOnHit;
         public Tancik(Vector<Gene> genes) {
+
+            this.genesRun = genes;
+            this.genesOnScanned = genes;
+            this.genesOnHit = genes;
             this.genes = genes;
+
         }
 
         public void run() {
 
-            for (Gene gene : genes) {
+            for (Gene gene : genesRun) {
```

Comment:
Kazdy `for` je treba jeste do cyklu `while`, aby metoda byla spustena vice nez jedenkrat.

Reply:
Run se zpouští sám opakovaně ne ?

Reply:
Pokud jsem se dobre dival na run v ostatnich tancich, tak dane metody, ktere se tam volaji jsou ve smycce. 
https://robocode.sourceforge.io/docs/robocode.dotnet/html/908fc832-ef42-a1a6-27a6-afbcb4fe85b6.htm